packages/atomic/src/components/atomic-search-interface/atomic-search-interface.tsx

```diff
@@ -257,18 +255,19 @@ export class AtomicSearchInterface {
   }
 
   private updateHash() {
-    this.updatingHash = true;
-    defer(() => (this.updatingHash = false));
-    const hash = this.urlManager.state.fragment;
-    window.location.hash = hash;
-  }
-
-  private onHashChange = () => {
-    if (this.updatingHash) {
+    if (this.urlManager.hasFragment(this.fragment)) {
```

Comment:
If `pushState` does not trigger a hash change, is this `if` statement still needed? If not, then we could simplify even more by removing `hasFragment` from the url manager's public api.

Reply:
- In my interaction, I do a previous or next in the browser
- The hash changes, triggering the event
- synchronize() is called with the new fragment
- In headless, that updates the state, and the listener is called (updateHash)
- We don't want to push state here, since the fragment is the same, we would have duplicate histories

In the other hand, pushState does not trigger a hash change, so there's no check in the onHashChange method.
Because we used `window.location.hash = hash;` before, we didn't have that problem, now at have to compare the state's fragment with the current url fragment before pushing

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/History/pushState

> the history.pushState() method adds an entry to the browser's session history stack.

> You don't have to change the URL if you don't want to. In contrast, setting window.location = "#foo"; only creates a new history entry if the current hash isn't #foo.

So changing location manually only added an history stack if the hash changed, but pushState will always add a stack, requiring a comparison of the hashes/fragments